AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1793,7 +1795,9 @@ - (BOOL)placeholderShouldPersist
 
 - (BOOL)_shouldHavePlaceholderLayer
 {
-  return (_placeholderEnabled && [self __implementsDisplay]);
+  ASDisplayNodeAssertMainThread();
+  CGSize calculatedSize = [self calculatedSize];
```

Comment:
Thanks for adding the assert!

What happens in the case you have an ASMapNode (or anything), and set its frame = Rect(5, 5, 100, 100) without measure: being called?  I believe calculatedSize will be zero, even though the element will be fully onscreen and prior to this diff would show a placeholder.